examples/deadline/All-In-AWS-Infrastructure-SEP/python/package/lib/sep_stack.py

```diff
@@ -73,46 +99,79 @@ def __init__(self, scope: Construct, stack_id: str, *, props: SEPStackProps, **k
             repository_installation_timeout=Duration.minutes(20)
         )
 
+        # The following code is used to demonstrate how to use the SEPConfigurationSetup if TLS is enabled.
+        host = 'renderqueue'
+        zone_name = 'deadline-test.internal'
+
+        # Internal DNS zone for the VPC.
+        dns_zone = PrivateHostedZone(
+            self,
+            'DnsZone',
+            vpc=vpc,
+            zone_name=zone_name
+        )
+
+        # NOTE: This certificate is also used by SEPConfigurationSetup construct below.
+        ca_cert = X509CertificatePem(
+            self,
+            'RootCA',
+            subject=DistinguishedName(
+                cn='SampleRootCA'
+            )
+        )
+
+        server_cert = X509CertificatePem(
+            self,
+            'RQCert',
+            subject=DistinguishedName(
+                cn=f'{host}.{dns_zone.zone_name}',
+                o='RFDK-Sample',
+                ou='RenderQueueExternal'
+            ),
+            signing_certificate=ca_cert
+        )
+
         render_queue = RenderQueue(
             self,
             'RenderQueue',
-            vpc=props.vpc,
+            vpc=vpc,
             version=recipes.version,
             images=recipes.render_queue_images,
             repository=repository,
             # TODO - Evaluate deletion protection for your own needs. This is set to false to
             # cleanly remove everything when this stack is destroyed. If you would like to ensure
             # that this resource is not accidentally deleted, you should set this to true.
-            deletion_protection=False
+            deletion_protection=False,
+            hostname=RenderQueueHostNameProps(
+                hostname=host,
+                zone=dns_zone
+            ),
+            traffic_encryption=RenderQueueTrafficEncryptionProps(
+                external_tls=RenderQueueExternalTLSProps(
+                    rfdk_certificate=server_cert
+                ),
+                internal_protocol=ApplicationProtocol.HTTPS
+            )
+        )
+
+        # Create the IAM Role for the spot fleet.
+        # Note if you already have a worker IAM role in your account you can use it instead.
+        fleet_role = Role(
```

Comment:
At first, I wanted to create this role inside of the `SEPSpotFleet` construct. But this created a circular dependency.
Also, I don't think we need to create a separate role for each SpotFleet we create, we can just reuse one. And finally, we need to allow the Render Queue to pass this role, so it seems better to create this role externally and then to pass it to the `SEPSpotFleet` construct.

Thoughts?

Reply:
Resolving. Moved the creation of this role back inside of SpotFleet construct following https://github.com/aws/aws-rfdk/pull/279#discussion_r571295524